### PR TITLE
PAYM-2893 Event JSON schema validation

### PR DESCRIPTION
**Pull request by dabeat**
#### :tophat: Problem
JSON schema validation needs to be done.

#### :pushpin: Solution
Provide alternative flow detached from current one, to make sure backwards compatibility is provided in the migration to the new flow.

#### :ghost: GIF
giphy

### CODE REVIEW

File: lib/events/event.ex [hunk 1]
```diff
@@ -46,6 +47,22 @@ defmodule ElixirTools.Events.Event do
...
   end
 
+  @spec publish(Event.t(), event_schema, [publish_opts]) :: return
+  def publish(event, schema, opts \\ []) do
+    adapter =
+      opts[:adapter] || Application.get_env(:pagantis_elixir_tools, ElixirTools.Events)[:adapter]
+
+    enveloped_event = add_envelope(event, opts)
+
+    with :ok <- validate_json_schema(enveloped_event, schema) do
```
Comment (Kitton) [inline lib/events/event.ex:57]: two think here:
* do we need `with` in this case?
* if `validate_json_schema` returns error do we create record and DB and send rollbar message?

Comment (dabeat) [reply lib/events/event.ex:57]: * Maybe we don't. I just duplicated current implementation to introduce the less possible number of code modifications. If we make sure we don't need it, I'll modify it in another PR.

* Yes, it has the same "impact" as the validations we are doing in the current flow to be deprecated. `EventHandler` is in charge of dealing with these errors.

File: lib/events/event_handler.ex [hunk 2]
```diff
@@ -78,7 +74,17 @@ defmodule ElixirTools.Events.EventHandler do
...
 
-    case event_module.publish(event) do
+    case event_module.publish_deprecated(event) do
+      {:error, reason} -> handle_error(event, reason, opts)
+      _ -> :ok
+    end
+  end
+
+  @spec publish_event_call(Event.t(), event_schema, [events_opt]) :: :ok | :error
+  def publish_event_call(event, schema, opts) do
```
Comment (Kitton) [inline lib/events/event_handler.ex:84]: looks like `publish_event_call` should be private

File: lib/events/event_handler.ex [hunk 3]
```diff
@@ -78,7 +74,17 @@ defmodule ElixirTools.Events.EventHandler do
   def publish_event_call(event, opts) do
```
Comment (Kitton) [inline lib/events/event_handler.ex:74]:
```suggestion
  defp publish_event_call(event, opts) do
```

Comment (dabeat) [reply lib/events/event_handler.ex:74]: I'll change, butt that is out of the scope of this PR and in any case is code to be removed when all the events are migrated to the new flow.

**Pull request merged**